### PR TITLE
re: `near-network`: Rewrite `local_edge_info` in `RoutingTableView`

### PR DESCRIPTION
**Pull request by pmnoxx**
We can make a few code simplifications in `RoutingTableView`.
We currently only keep local edges in `local_edges_info`, once of peers is `my_peer_id`, so there is no need to store a pair anymore.

We can change signature of `local_edges_info`
```
pub local_edges_info: HashMap<(PeerId, PeerId), Edge>,
```
to
```
pub local_edges_info: HashMap<PeerId, Edge>,
```

### CODE REVIEW

File: chain/network/src/routing/routing.rs [hunk 1]
```diff
@@ -194,8 +193,9 @@ impl RoutingTableView {
 
     pub fn remove_edges(&mut self, edges: &Vec<Edge>) {
         for edge in edges.iter() {
-            assert!(edge.key().0 == self.my_peer_id || edge.key().1 == self.my_peer_id);
-            self.local_edges_info.remove(edge.key());
+            if let Some(other_peer) = edge.other(&self.my_peer_id) {
+                self.local_edges_info.remove(other_peer);
```
Comment (mm-near) [inline chain/network/src/routing/routing.rs:197]: now we're going to silently ignore the edges that don't start from the current peer (before we were asserting).

if this was intended - please add the comment to the function

Comment (mm-near) [reply chain/network/src/routing/routing.rs:197]: or maybe rename to something like 'remove_edges_if_local' ?

File: chain/network/src/peer_manager/peer_manager_actor.rs [hunk 2]
```diff
@@ -1900,19 +1885,22 @@ impl PeerManagerActor {
...
-                            if edge.nonce() >= *nonce {
-                                self.local_peer_pending_update_nonce_request.remove(other);
+                if let Some(other_peer) = edge.other(&self.my_peer_id) {
+                    if edge.verify() {
+                        if self.routing_table_view.is_local_edge_newer(other_peer, edge.nonce()) {
+                            if let Some(nonce) =
+                                self.local_peer_pending_update_nonce_request.get(other_peer)
+                            {
+                                if edge.nonce() >= *nonce {
+                                    self.local_peer_pending_update_nonce_request.remove(other_peer);
```
Comment (mm-near) [inline chain/network/src/peer_manager/peer_manager_actor.rs:1895]: add comment to explain this case

**Review by mm-near — APPROVED** [review]

File: chain/network/src/routing/routing.rs (hunk 1 above)
Comment (pmnoxx) [reply chain/network/src/routing/routing.rs:197]: Ok, I renamed the method to `remove_local_edges` and I added panic!.

**Pull request merged**